.github/workflows/update_website.yml

```diff
@@ -0,0 +1,44 @@
+name: update website
+on:
+  push:
+    branches:
+      - 'main'
+  schedule:
+    - cron: '5 8 * * 0' # Evry Sunday 8.05
+  workflow_dispatch:
+
+jobs:
+  update_submodule:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
+
+      - name: update submodule
+        run: |
+            git submodule update --init --recursive
+            git submodule update --remote rohd
+
+      - name: check for changes
+        run: |
+            cd rohd/doc/user_guide
+            if git diff --quiet HEAD @{u}; then
+                echo "No changes in submodule."
+                exit 0
```

Comment:
Does this make this stage pass and so it continues to create the pull request?  Or the exit kicks it out?

Reply:
From what I see, it will proceed to open-up a pull request. Now, I think we might not need this as well, since `peter-evans/create-pull-request@v5` will auto detect changes and only open the pull request when there different with deploy-website.